base/abstractarray.jl

```diff
@@ -208,6 +208,13 @@ reshape(a::AbstractArray, dims::Int...) = reshape(a, dims)
 
 ## from general iterable to any array
 
+function copy!(dest, src)
+    for x in src
+        push!(dest, x)
```

Comment:
I feel like you'd have to `empty!(dest)` first to make this actually implement `copy!`, otherwise as written this is doing `append!`.